lua/nvim-treesitter-textobjects/shared.lua

```diff
@@ -1,12 +1,291 @@
 local api = vim.api
-
-local parsers = require "nvim-treesitter.parsers"
-local queries = require "nvim-treesitter.query"
-local ts_utils = require "nvim-treesitter.ts_utils"
-local ts = require "nvim-treesitter.compat"
+local ts = vim.treesitter
 
 local M = {}
 
+local EMPTY_ITER = function() end
+
+function M.set_jump()
+  vim.cmd "normal! m'"
+end
+
+---@param filetype string
+local function ft_to_lang(filetype)
+  local result = ts.language.get_lang(filetype)
+  if result then
+    return result
+  else
+    filetype = vim.split(filetype, "%.")[1]
+    assert(filetype, string.format("No language for filetype %s", filetype))
+    return ts.language.get_lang(filetype) or filetype
+  end
+end
+
+---@param bufnr integer
+---@return string lang
+function M.get_buf_lang(bufnr)
```

Comment:
If I remember correctly this might be possible even with pure nvim functions

Reply:
`vim.treesitter.language.get_lang(vim.bo.ft)`